docs/sample_log_config.yaml

```diff
@@ -18,13 +18,24 @@ filters:
 
 handlers:
     file:
-        class: logging.handlers.RotatingFileHandler
+        class: logging.handlers.TimedRotatingFileHandler
         formatter: precise
         filename: /var/log/matrix-synapse/homeserver.log
-        maxBytes: 104857600
-        backupCount: 10
+        when: midnight
+        backupCount: 3  # Does not include the current log file.
         filters: [context]
         encoding: utf8
+
+    # Default to buffering writes to log file for efficiency. This means that
+    # will be a delay for WARNING/INFO/DEBUG logs to get written, but ERROR
+    # logs will still be flushed immediately.
+    buffer:
```

Comment:
you need to do the `filters:[context]` thing here, otherwise we get the request ID at the point the logs get written to disk rateher than the point they were logged, which isn't that useful

Reply:
So I realised that the whole `filters: [context]` thing is a potential foot gun due to these sorts of issues. We could heavily comment in the default log config, but instead I've moved the logic into the code so that the filter automatically gets applied correctly.

Reply:
well I think that's a good idea, and I agree it makes this much less of a footgun; however:
 * lots of people will have the filter set up already: can you confirm that having it twice is harmless (I can't see why it wouldn't be)
 * we should document this: people will want to know whether they *have* to update their log configs, what happens if they don't, etc

Reply:
I think this might be worth making a separate PR - or at least giving it a separate changelog entry. It's a nice improvement and I don't know why we didn't do it sooner.

Reply:
Done https://github.com/matrix-org/synapse/pull/8051